private_sharing/api_views.py

```diff
@@ -458,3 +470,92 @@ def post(self, request):
                 data_file.delete()
 
         return Response({"ids": ids}, status=status.HTTP_200_OK)
+
+
+class ProjectCreateAPIView(APIView):
+    """
+    Create a project via API
+
+    Accepts project name and description as (required) inputs
+
+    A third input that should be provided is the first part of the redirect url;
+    this will get the new project's slug appended to it to form the new project's
+    oauth2 redirect url, eg <mysuperspiffydomain>/diyprojects/<new-project-slug>/complete/
+
+    The other required fields are auto-populated:
+    is_study:  set to False
+    leader:  set to member.name from oauth2 token
+    coordinator:  get from oauth2 token
+    is_academic_or_nonprofit: False
+    add_data:  false
+    explore_share:  false
+    short_description:  first 139 chars of long_description plus an elipse
+    active:  True
+    coordinator:  from oauth2 token
+    """
+
+    authentication_classes = (CustomOAuth2Authentication,)
+    permission_classes = (HasValidProjectToken,)
+
+    def get_short_description(self, long_description):
+        """
+        Return first 139 chars of long_description plus an elipse.
+        """
+        return "{0}…".format(long_description[0:139])
+
+    def post(self, request):
+        """
+        Take incoming json and create a project from it
+        """
+        project_creation_project = OAuth2DataRequestProject.objects.get(
+            pk=self.request.auth.pk
+        )
+
+        # If the first part of the redirect_url is provided, grab that, otherwise set
+        # to the project-creation-project's enrollment_url as a usable default
+        redirect_url_part = request.data.get("redirect-url-part", None)
```

Comment:
I think it's fine to make this required.

But I don't think it's a good idea to assume it should be based on the project slug. I think that's a cool idea! But the app using this API should be able to decide whether it wants that format or not, on its end.

So the field should probably be `redirect_url`.